src/MSBuild/Resources/Strings.resx

```diff
@@ -668,11 +668,13 @@ Copyright (C) Microsoft Corporation. All rights reserved.
   <data name="HelpMessage_31_RestoreSwitch" UESanitized="false" Visibility="Public">
     <value>  /restore[:True|False]
                      Runs a target named Restore prior to building
-                     other targets.  This is useful when your project
-                     tree requires packages to be restored before they
-                     can be built. Specifying /restore is the same as
-                     specifying /restore:True.  Use the parameter to
-                     override a value that comes from a response file.
+                     other targets and ensures the build uses the latest
+                     restore output instead of possibly cached output.
```

Comment:
Maybe 

> ensures the build uses the latest restored build logic.

? But I'm not sure that's actually clearer.

Reply:
I think it's more specific ("correct") but maybe not more helpful..

Reply:
Maybe something like

```
  /restore[:True|False]
                     Runs a target named Restore, resets the context,
                     and then runs other targets.  This is useful when
                     your project tree requires packages to be
                     restored before they can be built. Specifying
                     /restore is the same as specifying /restore:True.
                     Use the parameter to override a value that comes
                     from a response file.
                     (Short form: /r)
```

Reply:
> `resets the context`

I agree that this is more "correct".
However, my goal was to explicitly tell users that they should use it to get correct restore+build behavior.

Any ideas how to address both concerns?

```

```

Btw I also changed `… before they can be built.` => `... before it can be built.` in this PR since `project tree` is singular.

Reply:
LG2M